pkg/flow/internal/controller/component_node_manager.go

```diff
@@ -43,15 +43,25 @@ func (m *ComponentNodeManager) createComponentNode(componentName string, block *
 }
 
 // getCustomComponentConfig is used by the custom component to retrieve its template and the customComponentRegistry associated with it.
-func (m *ComponentNodeManager) getCustomComponentConfig(componentName string) (ast.Body, *CustomComponentRegistry, error) {
+func (m *ComponentNodeManager) getCustomComponentConfig(namespace string, componentName string) (ast.Body, *CustomComponentRegistry, error) {
 	m.mut.Lock()
 	defer m.mut.Unlock()
 
-	template, customComponentRegistry := findLocalDeclare(m.customComponentReg, componentName)
+	var (
+		template                ast.Body
+		customComponentRegistry *CustomComponentRegistry
+	)
+
+	if namespace == "" {
+		template, customComponentRegistry = findLocalDeclare(m.customComponentReg, componentName)
+	} else {
+		template, customComponentRegistry = findImportedDeclare(m.customComponentReg, namespace, componentName)
+	}
 
 	if customComponentRegistry == nil || template == nil {
-		return nil, nil, fmt.Errorf("custom component config not found in the registry, componentName: %s", componentName)
+		return nil, nil, fmt.Errorf("custom component config not found in the registry, namespace: %s, componentName: %s", namespace, componentName)
```

Comment:
nit: I encourage using `%q` over `%s` for strings wherever possible; `%q` produces a quoted string and makes it much easier to notice when an empty string is being printed. 

(sorry I missed this before)